### ISSUE TITLE
what is thi-startup about (what[s] are we building)

### ISSUE THREAD

**Issue by Joe-Degs** [thi-startup/RFC#1]
The question I keep asking myself if what we are doing exactly, is there one thing (tool, system, framework or library) we are building or is it something generic. I for one think its generic with varying focus points are as we move along. But I think there was a birthing story explaining why we are taking on this fun challenge in the first place, and I think that birthing story goes like this;

### the story
Joe is a guy that wants to learn how to build sophisticated distributed systems, he is not really well so he wants to learn how to build these distributed systems things on actual hardware systems that are distributed or at least a simulation of it. He has tried traditional virtual machines but has discovered that jeez they are hard to setup most of the time and require work bring up and down because you have to run actual operating systems on top of them. there is tooling to make running Vms hassle free but he has discovered that every tool is finicky and when Joe hates finicky, he hates it. 
He tried docker and didn't really like all the work you have to do to get operating systems up and running on those things (he doesn't exactly like using docker too, but he never talks about it). One thing that really stuck if how its easy to bring docker things up and down and how it all just works without you thinking about it, just like the way cloud providers allow you to just bring up vms with operating systems within minutes. Joe wants what the cloud people provide but doesn't have the money to pay cloud people for their services.

Then one time his friend, let's call him **bxffour**, called Joe one evening whilst he was on a walk thinking about what the point of anything really is (you know, one of those long depressing nights). And he told joe about this new fascinating thing he had been trying to use called **Firecracker**. Joe wasn't exactly interested in listening to another jibby jabber about another tech thing, but if you know bxffour you will understand why he shut up and pretended to listen, and then again and again for about a month (because it was all bxffour talked about anymore).

Joe tried his best but if you are exposed to jibby jabber for that long, a bit of it tends to sip into your brain. but then it lit a bulb in joe's brain, that is exactly what I want. A tool to setup ephemeral virtual machines in seconds (or very small minutes) on your localhost, do things with them and then bring them down again (exactly like you would do on AWS or GCP) -- turns out that was what bxffour was jibby jabbering about all this while. Now Joe is excited, he wants to build something like this, so he can continue learning to do distributed systems, setting up and bringing vms like he is on a cloud provider.

Right now there is Firecracker taking care of the virtual machine management part of the work, but it is fairly new and there is little to no tooling available for bootstrapping these vms on local machines. Joe and bxffour want to take up the task of building out the thing that makes it easy to use firecracker on localhost and this is alot of work. it scares Joe but he really need this so he doesn't exactly have a choice.

**Comment by Joe-Degs** [comment]
@bxffour  what do you think? I was writing something short but started and couldn't stop, sorry...

**Comment by bxffour** [comment]
> @bxffour what do you think? I was writing something short but started and couldn't stop, sorry...

This bx 4 guy seems like a really cool guy. Misinformation aside. But I couldn't word it any better. It's perfect

**Comment by bxffour** [comment]
So essentially the question still goes unanswered. What are we building? Well one thing we know for sure is, we are building tools and libraries around firecracker but that should just be one cog in a possibly large machine. For example we might want to do networking stuff that has nothing to do with firecracker exactly but would be useful for what we might want to do. So basically we build systems and tools to make working with systems not stressful. I mean we're running out of hairs to tear out already. So right now it's tooling around firecracker, tomorrow it might be a spaceship to mars, who knows? God knows we don't so we shouldn't limit ourselves.

**Comment by Joe-Degs** [comment]
yes there might be different/varying points of focus in the future but the overarching theme is that we want infrastructure to build or simulate building complex distributed systems. And right now, the point of focus is building tooling that allows us to bend firecracker to our will to do our bidding.

**Comment by bxffour** [comment]
> yes there might be different/varying points of focus in the future but the overarching theme is that we want infrastructure to build or simulate building complex distributed systems. And right now, the point of focus is building tooling that allows us to bend firecracker to our will to do our bidding.

Ladies and Gentlemen.. Let us Cook!
